src/main/java/com/siit/hospital_manager/config/SecurityConfig.java

```diff
@@ -23,8 +23,9 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                 .authorizeHttpRequests()
                 .requestMatchers("/", "/public", "/api-docs/**", "/swagger-ui/**",
                         "/actuator/**", "/mvc/patient/create", "/validationError.html", "/error",
-                        "/favicon.ico", "/mvc/patient/submitCreatePatientForm", "/entityExistsError.html", "/static/**").permitAll()
-                .requestMatchers("/dashboard/**", "/dashboard", "/appointment/**").hasAnyRole("PATIENT", "ADMIN")
+                        "/favicon.ico", "/mvc/patient/submitCreatePatientForm/**", "/mvc/patient/viewAll", "/mvc/doctor/viewAll",
+                        "/mvc/patient/medicalHistory/**", "/mvc/patient/treatmentPlan" , "/entityExistsError.html", "/static/**").permitAll()
+                .requestMatchers("/dashboard/**", "/dashboard", "/appointment/**").hasAnyRole("PATIENT", "ADMIN", "DOCTOR")
```

Comment:
access control could be better defined, specifying who can acces what. for example viewAll doctors is in a permitAll request matcher, which is a security risk